File: Java/Tests/jdi-uitest-wintests/src/test/java/com/epam/jdi/uitests/testing/unittests/tests/composite/SearchTests.java
package com.epam.jdi.uitests.testing.unittests.tests.composite;

import com.epam.jdi.uitests.testing.unittests.InitTests;
import org.testng.annotations.BeforeMethod;
import org.testng.annotations.Test;

import java.lang.reflect.Method;

import static com.epam.jdi.uitests.core.preconditions.PreconditionsState.isInState;
import static com.epam.jdi.uitests.testing.unittests.enums.Preconditions.HOME_PAGE;
import static com.epam.jdi.uitests.testing.unittests.pageobjects.EpamJDIApplication.header;
import static com.epam.jdi.uitests.testing.unittests.pageobjects.EpamJDIApplication.supportPage;

/**
 * Created by Dmitry_Lebedev1 on 10/15/2015.
 */
public class SearchTests extends InitTests {
    @BeforeMethod
    public void before(final Method method) {
        isInState(HOME_PAGE, method);
    }

    @Test
    public void fillTest() {
        header.search.find("something");
        supportPage.checkOpened();
    }
}

File: C#.Net/Tests/JDI UIWebTests/JDI UIWebTests/Tests/Composite/SearchTests.cs
using NUnit.Framework;
using static Epam.JDI.Core.Settings.JDISettings;
using static JDI_UIWebTests.UIObjects.TestSite;

namespace JDI_UIWebTests.Tests.Composite
{
    public class SearchTests
    {

        [SetUp]
        public void SetUp()
        {
            Logger.Info("Navigating to Home page.");
            HomePage.Open();
            HomePage.CheckTitle();
            HomePage.IsOpened();
            Logger.Info("Setup method finished");
            Logger.Info("Start test: " + TestContext.CurrentContext.Test.Name);
        }

        [Test]
        public void FillTest()
        {
            Header.Search.SearchButton.Click();
            Header.Search.Find("something");
            Header.Search.SearchButonActive.Click();
            SupportPage.CheckOpened();
        }

    }
}
